### PR TITLE
dump: support new where sql generator

### PR DESCRIPTION
**Pull request by lichunzhu**
### What problem does this PR solve? 
close https://github.com/pingcap/dumpling/issues/259

### What is changed and how it works?
support new where sql generator.

### Check List 

Tests 

 - Unit test
 - Integration test
 - Manual test (add detailed scripts or steps below)

Side effects

 - Increased code complexity

Related changes

 - Need to cherry-pick to the release branch
 - Need to update the documentation
 
### Release note

- No release note

### CODE REVIEW

File: v4/export/sql.go [hunk 1]
```diff
@@ -570,25 +570,98 @@ func buildSelectField(db *sql.Conn, dbName, tableName string, completeInsert boo
 	return "*", len(availableFields), nil
 }
 
-func buildWhereClauses(handleColNames, handleVals []string) []string {
-	if len(handleVals) == 0 {
+func buildWhereClauses(handleColNames []string, handleVals [][]string) []string {
```
Comment (kennytm) [inline v4/export/sql.go:573]: please add a unit test for this function.

Comment (lichunzhu) [reply v4/export/sql.go:573]: added in https://github.com/pingcap/dumpling/pull/260/commits/3959b53251a2c2afdb0cc0ad5b46d1fe69244789

**Comment by lichunzhu** [comment]
cc @tangenta

**Comment by lichunzhu** [comment]
/hold
I will do a test on the v5.0 cluster and paste the result here later.

**Review by leoppro — APPROVED** [review]

**Pull request merged**